src/aws-cpp-sdk-core/source/utils/crypto/PrecalculatedHash.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+
+#include <aws/core/utils/crypto/PrecalculatedHash.h>
+
+using namespace Aws::Utils::Crypto;
+
+PrecalculatedHash::PrecalculatedHash(const Aws::String &mHash): m_hash(ByteBuffer((unsigned char*)mHash.c_str(),mHash.length())){
+m_hash_string = mHash;
+}
+
+PrecalculatedHash::~PrecalculatedHash() = default;
+
+HashResult PrecalculatedHash::Calculate(const Aws::String& str)
+{
+    AWS_UNREFERENCED_PARAM(str);
+    return m_hash;
+}
+
+HashResult PrecalculatedHash::Calculate(Aws::IStream& stream)
+{
+    AWS_UNREFERENCED_PARAM(stream);
+    return m_hash;
```

Comment:
do we need `m_hash` at all anymore? imo whenever we want to return a byte buffer we should be calling `HashingUtils::Base64Decode(m_hashString)`  that way we only are keeping track of one member variable and the constructor becomes a lot cleaner as we are only using one string param and not casting it to a byte array.

also it might be incorrect as we would be creating a byte buffer of the base54 encoding and not the actual hash result